src/Module/AllFriends.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Friendica\Module;
+
+use Friendica\BaseModule;
+use Friendica\Content\ContactSelector;
+use Friendica\Content\Pager;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Model;
+use Friendica\Network\HTTPException;
+use Friendica\Util\Proxy as ProxyUtils;
+
+class AllFriends extends BaseModule
+{
+	public static function content()
+	{
+		$app = self::getApp();
+
+		if (!local_user()) {
+			throw new HTTPException\ForbiddenException();
+		}
+
+		$cid = 0;
+
+		// @TODO: Replace with parameter from router
+		if ($app->argc > 1) {
+			$cid = intval($app->argv[1]);
+		}
+
+		if (!$cid) {
+			throw new HTTPException\BadRequestException(L10n::t('Invalid contact.'));
+		}
+
+		$uid = $app->user['uid'];
+
+		$contact = Model\Contact::getContactForUser($cid, local_user(), ['name', 'url', 'photo', 'uid', 'id']);
+
+		if (empty($contact)) {
+			throw new HTTPException\BadRequestException(L10n::t('Invalid contact.'));
+		}
+
+		$app->page['aside'] = "";
+		Model\Profile::load($app, "", 0, Model\Contact::getDetailsByURL($contact["url"]));
+
+		$total = Model\GContact::countAllFriends(local_user(), $cid);
+
+		$pager = new Pager($app->query_string);
+
+		$friends = Model\GContact::allFriends(local_user(), $cid, $pager->getStart(), $pager->getItemsPerPage());
+		if (empty($friends)) {
+			return L10n::t('No friends to display.');
+		}
+
+		$id = 0;
+
+		$entries = [];
+		foreach ($friends as $friend) {
+			//get further details of the contact
+			$contactDetails = Model\Contact::getDetailsByURL($friend['url'], $uid, $friend);
+
+			$connlnk = '';
+			// $friend[cid] is only available for common contacts. So if the contact is a common one, use contact_photo_menu to generate the photoMenu
+			// If the contact is not common to the user, Connect/Follow' will be added to the photo menu
+			if ($friend['cid']) {
+				$friend['id'] = $friend['cid'];
+				$photoMenu = Model\Contact::photoMenu($friend);
+			} else {
+				$connlnk = $app->getBaseURL() . '/follow/?url=' . $friend['url'];
+				$photoMenu = [
+					'profile' => [L10n::t("View Profile"), Model\Contact::magicLink($friend['url'])],
+					'follow'  => [L10n::t("Connect/Follow"), $connlnk]
+				];
+			}
+
+			$entry = [
+				'url'          => Model\Contact::magicLink($friend['url']),
```

Comment:
Please use `Model\Contact::magicLinkbyId` instead since we have the contact id in `$friend`.